url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/AgentConnection.java

```diff
@@ -0,0 +1,1042 @@
+			dumpstruct();
+			return true;
+		}
+
+		if (line.equals("w")){
+			this.agentName="CMD";
+			out.println(String.format("%d,%d,%d\n", p.coreState.getLoggedAgentList().size(),p.coreState.getLoggedAgentList().getFreeAgentCount(), p.coreState.getLoggedAgentList().getTalkingAgentCount()));
+			return true;
+		}
+		if (line.equals("nw")){
+			this.agentName="CMD";
+			for (RealQueue q:p.coreState.getQueueList().getValues()){
+					out.println(String.format("%s %d %d %d %d %d %d",q.getName(), q.getWaitCount(), q.getTalkingAgentCount(), 
+							q.getFreeAgentCount(), q.getPausedAgentCount() ,q.getPausedAgentCount(PauseType.ID_DEFAULT) ,q.getAgentList().size()));
+				}			
+			return true;
+		}
+		if (line.equals("reload")){
+			this.agentName="CMD";
+			ReadConfig c = new ReadConfig(p, p.c.configFile);
+			if (CCproxy.checkLicense(c.licensedAgents, c.licenseKey, c.licenseExpires)){
+				p.c=c;
+				log.info("Config reload OK");
+				out.println("Config reload OK");
+			} else
+			{
+				log.info("Config reload FAILED");
+				out.println("Config reload FAILED");
+			};
+			return true;
+		}
+		if (line.equals("status")){
+			this.agentName="CMD";
+			out.println("Config file: " + p.c.configFile);
+			out.println("Version: " + CCproxy.version);
+			out.println("Licensed agents: " + p.c.licensedAgents);
+			out.println("License expires: " + p.c.licenseExpires);
+			
+			return true;
+		}
+		if (line.startsWith("pause")){
+			// pause agent '302' value 'true' type 'wrap-up'
+			this.agentName="CMD";
+			Matcher m = Pattern.compile("pause agent '([^']*)' value '([^']*)' type '([^']*)'.*").matcher(line);
+			if (!m.matches()) return true;
+			String agent = m.group(1);
+			String exten;
+			try {
+				exten = p.coreState.findAgent(new RealAgent(agent,"","")).getExtension();
+			} catch (AgentNotFoundException e) {
+				out.println("ERR");
+				return true;
+			}
+			String str="EVENT [ProxyClient], Action [Pause], Value ["+m.group(2)+"], Type ["+m.group(3)+"], Agent ["+agent+"], Exten ["+exten+"]";
+			log.info(str);
+			p.gadajAsterisk.sendMessage(this, str);
+			out.println("OK");
+			return true;
+		}
+		if (line.startsWith("allowed pausetypes agent")){
+			//allowed pausetypes agent '304' values 'przerwa','zadanie służbowe'
+			this.agentName="CMD";
+			Matcher m = Pattern.compile("allowed pausetypes agent '([^']*)' values (.*)").matcher(line);
+			if (!m.matches()) return true;
+			String agent = m.group(1);
+			String pausetypestr = m.group(2);
+			RealAgent ra;
+			try {
+				ra = p.coreState.findAgent(new RealAgent(agent,"",""));
+			} catch (AgentNotFoundException e) {
+				out.println("ERR");
+				return true;
+			}
+			ArrayList<String> pausetypes=null;
+			if (! pausetypestr.equals("all")){
+				pausetypes=new ArrayList<String>();
+				if (! pausetypestr.equals("none")){
+					Matcher ma=Pattern.compile("'([^']*)'").matcher(pausetypestr);
+					while (ma.find()){
+						pausetypes.add(ma.group(1));
+					}
+				}
+			}	
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 626:634, 655:663.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.